src/app/routers/schema_router.py

```diff
@@ -59,11 +60,18 @@ async def get_schema_from_bucket(
 
     QueryParameterValidatorService.validate_schema_version_parses(version)
 
-    bucket_schema_filename = (
-        schema_firebase_repository.get_schema_metadata_bucket_filename(
-            survey_id, version
+    if version is None:
+        bucket_schema_filename = (
+            schema_firebase_repository.get_latest_schema_metadata_bucket_filename(
+                survey_id
+            )
+        )
+    else:
+        bucket_schema_filename = (
+            schema_firebase_repository.get_schema_metadata_bucket_filename(
+                survey_id, version
+            )
```

Comment:
I think this if/else logic should belong in a service. Determining what function should be called based on if the version is populated is business logic and shouldn't be the responsibility of the web layer